src/gui/dialogues/firstrundialog.cpp

```diff
@@ -33,149 +36,144 @@ FirstRunDialog::~FirstRunDialog()
-        }
-        ASettings::write("setup/setup_complete", true);
-        aDB()->disconnect(); // reset db connection to refresh layout after initial setup.
-        aDB()->connect();
-        auto pilot = APilotEntry(1);
-        pilot.setData(data);
-        if (aDB()->commit(pilot)) {
-            qApp->quit();
-            QProcess::startDetached(qApp->arguments()[0], qApp->arguments());
-        } else {
-            QMessageBox message_box(this);
-            message_box.setText("Errors have ocurred creating the database. Without a working database The application will not be usable.");
-        }
+    ASettings::write(ASettings::UserData::LastName, ui->lastnameLineEdit->text());
+    ASettings::write(ASettings::UserData::FirstName, ui->firstnameLineEdit->text());
+    ASettings::write(ASettings::UserData::EmployeeID, ui->employeeidLineEdit->text());
+    ASettings::write(ASettings::UserData::Phone, ui->phoneLineEdit->text());
+    ASettings::write(ASettings::UserData::Email, ui->emailLineEdit->text());
+
+    ASettings::write(ASettings::FlightLogging::Function, ui->functionComboBox->currentText());
+    ASettings::write(ASettings::FlightLogging::Approach, ui->approachComboBox->currentText());
+    ASettings::write(ASettings::FlightLogging::NightLogging, ui->nightComboBox->currentIndex());
+    ASettings::write(ASettings::FlightLogging::LogIFR, ui->rulesComboBox->currentIndex());
+    ASettings::write(ASettings::FlightLogging::FlightNumberPrefix, ui->prefixLineEdit->text());
+    ASettings::write(ASettings::FlightLogging::NumberTakeoffs, 1);
+    ASettings::write(ASettings::FlightLogging::NumberLandings, 1);
+    ASettings::write(ASettings::FlightLogging::PopupCalendar, true);
+    ASettings::write(ASettings::FlightLogging::PilotFlying, true);
+
+    QMap<QString, QVariant> data;
+    ASettings::write(ASettings::UserData::DisplaySelfAs, ui->aliasComboBox->currentIndex());
+    data.insert(ASettings::stringOfKey(ASettings::UserData::LastName), ui->lastnameLineEdit->text());
+    data.insert(ASettings::stringOfKey(ASettings::UserData::FirstName), ui->firstnameLineEdit->text());
+    data.insert(ASettings::stringOfKey(ASettings::UserData::Alias), "self");
+    data.insert(ASettings::stringOfKey(ASettings::UserData::EmployeeID), ui->employeeidLineEdit->text());
+    data.insert(ASettings::stringOfKey(ASettings::UserData::Phone), ui->phoneLineEdit->text());
+    data.insert(ASettings::stringOfKey(ASettings::UserData::Email), ui->emailLineEdit->text());
+
+    auto db_fail_msg_box = QMessageBox(QMessageBox::Critical, QStringLiteral("Database setup failed"),
+                                       QStringLiteral("Errors have ocurred creating the database."
+                                                      "Without a working database The application will not be usable."));
+    // [G]: Im abit confused on the logic here
+    // why do you write setup complete twice?
+    if (!setupDatabase()) {
```

Comment:
This returns false. Tracing back the problem lead to error in ADataBaseSetup::createSchemata look for comment on that function